testing-suite/staking-v4/scenarios/PR_6114.py

```diff
@@ -0,0 +1,181 @@
+
+
+def main():
+    print("Happy testing")
+
+
+def test_PR_6114():
+
+    # === PRE-CONDITIONS ==============================================================
+    AMOUNT_TO_MINT = "10000" + "000000000000000000"
+
+    _A = Wallet(Path("./wallets/walletKey_1.pem"))
+    key_1 = ValidatorKey(Path("./validatorKeys/validatorKey_1.pem"))
+    key_2 = ValidatorKey(Path("./validatorKeys/validatorKey_2.pem"))
+    all_keys = [key_1, key_2]
+
+    # check if minting is successful
+    assert "success" in _A.set_balance(AMOUNT_TO_MINT)
+
+    # add some blocks
+    response = add_blocks(5)
+    assert "success" in response
+    time.sleep(0.5)
+
+    # check balance
+    assert _A.get_balance() == AMOUNT_TO_MINT
+
+    # add keys to protocol. This way staked keys will not go to jail.
+    assert "success" in add_key(all_keys)
+
+    # move to epoch
+    assert "success" in add_blocks_until_epoch_reached(5)
+
+    # === STEP 1 ==============================================================
+    # 1) In epoch 5 stake with A that 2 keys  - will go in auction
+
+    tx_hash = stake(_A, all_keys)
+
+    # check if tx is success
+    assert "success" in add_blocks_until_tx_fully_executed(tx_hash)
+
+    # check that nodes are staked
+    for key in all_keys:
+        assert "staked" in key.get_status(_A.public_address())
+
+    # move 2 blocks before calling validator/statistics
+    assert "success" in add_blocks(2)
+    time.sleep(1)
+
+    # check that nodes are in auction
+    for key in all_keys:
+        assert "auction" in key.get_state()
+
+    # === STEP 2 ==============================================================
+    # 2) Add blocks and epochs until 1 of this 2 keys are eligible - let's call it epoch X
+
+    eligible_key = add_blocks_until_key_eligible(all_keys)
+
+    current_epoch = proxy_default.get_network_status().epoch_number
+    print("Eligible key: ", eligible_key.public_address())
+    print("Current epoch:", current_epoch)
+
+    # === STEP 3 ==============================================================
+    # 3) unStake the eligible key in epoch X
+
+    tx_hash = unStake(_A, eligible_key)
+
+    # check if tx is success
+    assert "success" in add_blocks_until_tx_fully_executed(tx_hash)
+
+    current_epoch = proxy_default.get_network_status().epoch_number
+    print("Key unstaked in epoch:", current_epoch)
+
+    # check if key is now unStaked on getBlsKeyStatus
+    assert "unStaked" in eligible_key.get_status(_A.public_address())
+
+    # eligible_key becomes un_staked_key to be easier to read the test
+    un_staked_key = eligible_key
+
+    # === STEP 4 ==============================================================
+    # 4) call auction list api after 2 blocks and:
+
+    # wait 1 sec and move to the last block of epoch before saving all qualified and unqualified keys
+    time.sleep(1)
+    assert "success" in add_blocks_until_last_block_of_current_epoch()
```

Comment:
You don't need to move that much, just 1,2 blocks, but can be kept as it is

Reply:
Yes. Let's keep it for diversification. Api calls on latest block of the epoch. Can't hurt.